pkg/http/client/oauth2_token_source.go

```diff
@@ -0,0 +1,51 @@
+package client
+
+import (
+	"context"
+	"net/http"
+
+	configuration "github.com/buildbarn/bb-storage/pkg/proto/configuration/http/client"
+	"github.com/buildbarn/bb-storage/pkg/util"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// NewTokenSourceFromConfiguration uses the given configuration to create a
+// token source for HTTP clients.
+func NewTokenSourceFromConfiguration(oauthConfig *configuration.OAuth2Configuration) (oauth2.TokenSource, error) {
```

Comment:
NewOAuth2TokenSourceFromConfiguration()? Probably a bit more accurate, given that this lives in pkg/http/client, which doesn't have "oauth2" in its name.